local-cli/server/findSymlinksPaths.js

```diff
@@ -6,7 +6,28 @@ module.exports = function findSymlinksPaths(lookupFolder) {
   const folders = fs.readdirSync(lookupFolder);
   const resolvedSymlinks = folders.map(folder => path.resolve(lookupFolder, folder))
     .filter(folderPath => fs.lstatSync(folderPath).isSymbolicLink())
-    .map(symlink => path.resolve(process.cwd(), fs.readlinkSync(symlink)));
+    .map(symlink => {
+      const symlinkCache = new Set();
```

Comment:
We can easily detect symlink recursion just by caching each resolved symlink.
Though we could improve the error message by (1) keeping track of the resolution order and (2) printing the symlinks resolved in between the duplicate symlink and the first time it was resolved.